internal/bminventory/inventory.go

```diff
@@ -919,6 +919,25 @@ func (b *bareMetalInventory) DeregisterClusterInternal(ctx context.Context, para
 		log.Warnf("failed to delete DNS record sets for base domain: %s", cluster.BaseDNSDomain)
 	}
 
+	for _, h := range cluster.Hosts {
+		infraEnv, err2 := common.GetInfraEnvFromDB(b.db, h.InfraEnvID)
```

Comment:
Consider adding a helper function for this block.

Reply:
Done